### PR TITLE
ceph: set pg_num_min for rgw metadata pools

### LINKED ISSUE #5176: PG count for rgw metadata pool still at 32 with pg_autoscaler
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
With #5091 rgw metadata pool are created with 8 pgs by default based on rgw_rados_pool_pg_num_min parameter.
But with pg_autoscaler plugin activated the autoscaler immediately increase the size of those pool to it default PG_NUM_MIN of 32.

**Expected behavior:**
In order to avoid this rook should set pg_num_min on those pool to the value of rgw_rados_pool_pg_num_min

**How to reproduce it (minimal and precise):**

- Create a cluster on v14.2.8 or newer
- Activate pg atoscaer plugin
- Create a ceph object store
- Check number of pgs per metadata pool (should be 32)

### PR DESCRIPTION
**Pull request by ashangit**
Since e396379 rook create rgw metadata pools with 8 PGs instead of
32. But the pg_autoscaler increase it back to 32 as the pg_num_min parameter
is not set on those pools.

Signed-off-by: n.fraison <n.fraison@criteo.com>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #5176 

**Checklist:**

- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [ ] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](https://github.com/rook/rook/blob/master/CONTRIBUTING.md#comments)
- [ ] Add the flag for skipping the CI if this PR does not require a build. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for more details.

### CODE REVIEW

File: pkg/operator/ceph/object/objectstore.go [hunk 1]
```diff
@@ -315,6 +315,12 @@ func createSimilarPools(context *Context, pools []string, poolSpec cephv1.PoolSp
 					}
 				}
 			}
+			if pgCount != ceph.DefaultPGCount {
+				var err = ceph.SetPoolProperty(context.Context, context.ClusterName, name, "pg_num_min", pgCount)
```
Comment (leseb) [inline pkg/operator/ceph/object/objectstore.go:319]: Wyy re-initializing a new var here? You could just re-use `err`.

File: pkg/operator/ceph/object/objectstore.go [hunk 2]
```diff
@@ -315,6 +315,12 @@ func createSimilarPools(context *Context, pools []string, poolSpec cephv1.PoolSp
...
+				if err != nil {
+					return errors.Errorf("failed to set pg_num_min on pool %s to %s", name, pgCount)
```
Comment (leseb) [inline pkg/operator/ceph/object/objectstore.go:321]:
```suggestion
					return errors.Wrapf(err, "failed to set pg_num_min on pool %q to %q", name, pgCount)
```

File: pkg/operator/ceph/config/monstore.go [hunk 3]
```diff
@@ -71,7 +72,7 @@ func (m *MonStore) Get(who, option string) (string, error) {
 	if err != nil {
 		return "", errors.Wrapf(err, "failed to get config setting %q for user %q", option, who)
 	}
-	return string(out), nil
+	return strings.TrimSpace(string(out)), nil
```
Comment (leseb) [inline pkg/operator/ceph/config/monstore.go:75]: What is this fixing? It should be a separate commit.

Comment (ashangit) [reply pkg/operator/ceph/config/monstore.go:75]: Without this the ceph osd pool set command failed

Comment (leseb) [reply pkg/operator/ceph/config/monstore.go:75]: Do you have the error?

Comment (ashangit) [reply pkg/operator/ceph/config/monstore.go:75]: 2020-04-03 11:36:57.428631 I | exec: Error EINVAL: error parsing int value '8
2020-04-03 11:36:57.428895 E | op-object: failed to create or update object store toto. failed to create pools: failed to create object pools: failed to create metadata pools: failed to set pg_num_min on pool toto.rgw.control to 8

Comment (leseb) [reply pkg/operator/ceph/config/monstore.go:75]: It's interesting that we never caught this, please make it a separate commit since it's a different issue. Thanks.

Comment (leseb) [reply pkg/operator/ceph/config/monstore.go:75]: Can you share your CR file?

Comment (ashangit) [reply pkg/operator/ceph/config/monstore.go:75]: The pool creation works fine with that parameter but not the pool set command, this is probably why you never caught it:
Here is the CR file: https://gist.github.com/ashangit/77619d7ebbff8fc104b906e9916a4951

Comment (ashangit) [reply pkg/operator/ceph/config/monstore.go:75]: And here is the complete log:
```
2020-04-06 15:32:37.951382 I | op-object: creating object store test-1 in namespace rook-ceph
2020-04-06 15:32:37.959705 I | op-object: Gateway service running at None:9002
2020-04-06 15:32:37.959818 I | exec: Running command: ceph config get mon. rgw_rados_pool_pg_num_min --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/410161417
2020-04-06 15:32:38.578074 I | op-config: Get out 8
 out
2020-04-06 15:32:38.578236 I | exec: Running command: ceph osd pool get test-1.rgw.control all --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/448665812
2020-04-06 15:32:39.051246 I | exec: Error ENOENT: unrecognized pool 'test-1.rgw.control'
2020-04-06 15:32:39.051590 I | exec: Running command: ceph osd crush rule create-replicated test-1.rgw.control default rack --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/640261667
2020-04-06 15:32:40.501071 I | exec: Running command: ceph osd pool create test-1.rgw.control 8
 replicated test-1.rgw.control --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/229208358
2020-04-06 15:32:41.503653 I | exec: pool 'test-1.rgw.control' created
2020-04-06 15:32:41.503966 I | exec: Running command: ceph osd pool set test-1.rgw.control size 3 --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/111100493
2020-04-06 15:32:42.560851 I | exec: set pool 360 size to 3
2020-04-06 15:32:42.561181 I | exec: Running command: ceph osd pool application enable test-1.rgw.control rook-ceph-rgw --yes-i-really-mean-it --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/712489479
2020-04-06 15:32:43.542700 I | exec: enabled application 'rook-ceph-rgw' on pool 'test-1.rgw.control'
2020-04-06 15:32:43.542906 I | cephclient: creating replicated pool test-1.rgw.control succeeded, buf: 
2020-04-06 15:32:43.543050 I | exec: Running command: ceph osd pool set test-1.rgw.control pg_num_min 8
 --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/249563089
2020-04-06 15:32:44.005018 I | exec: Error EINVAL: error parsing int value '8
2020-04-06 15:32:44.005271 E | op-object: failed to create or update object store test-1. failed to create pools: failed to create object pools: failed to create metadata pools: failed to set pg_num_min on pool "test-1.rgw.control" to "8\n": failed to set pool property pg_num_min on pool test-1.rgw.control: exit status 22
```
As you can see the pool creation works fine despite the trailing space but the pool set one failed
